src/multi/bmp180.cpp

```diff
@@ -0,0 +1,305 @@
+               average_uncompensated_pressure = 0;
+
+  // loop to wait for conversion to complete
+  int conversion_finished;
+  for (int i = 0; i < sample_amount; i++) {
+    conversion_finished = 0;
+    hal::write(*m_i2c, m_address, sensor_config_buffer, hal::never_timeout());
+    while (conversion_finished == 0) {
+      conversion_finished = conversion_status(m_i2c, m_address);
+    }
+
+    // grab the raw data from the conversion and store it
+    auto uncompensated_data_buffer =
+      hal::write_then_read<3>(*m_i2c,
+                              m_address,
+                              uncompensated_output_data_register(),
+                              hal::never_timeout());
+    std::int32_t uncompensated_pressure =
+      ((uncompensated_data_buffer[0] << 16) |
+       (uncompensated_data_buffer[1] << 8) | uncompensated_data_buffer[2]) >>
+      (8 - oversampling_setting);
+    // accumulate the raw samples together to later find average
+    summed_uncompensated_pressures += uncompensated_pressure;
+  }
+  average_uncompensated_pressure =
+    summed_uncompensated_pressures / sample_amount;
+
+  // convert the raw pressure data into Pascals in steps of 1Pa
+  variables.b6 = temperature_data.b5 - 4000;
+  variables.x1 =
+    (m_calibration_data.b2 * ((variables.b6 * variables.b6) >> 12)) >> 11;
+  variables.x2 = (m_calibration_data.ac2 * variables.b6) >> 11;
+  variables.x3 = variables.x1 + variables.x2;
+  variables.b3 =
+    (((static_cast<std::int32_t>(m_calibration_data.ac1) * 4 + variables.x3)
+      << oversampling_setting) + 2) >> 2;
+  variables.x1 = (m_calibration_data.ac3 * variables.b6) >> 13;
+  variables.x2 =
+    (m_calibration_data.b1 * ((variables.b6 * variables.b6) >> 12)) >> 16;
+  variables.x3 = ((variables.x1 + variables.x2) + 2) >> 2;
+  variables.b4 = (m_calibration_data.ac4 *
+                  static_cast<std::uint32_t>(variables.x3 + 32768)) >> 15;
+  variables.b7 = (static_cast<std::uint32_t>(average_uncompensated_pressure) -
+                  variables.b3) * (50000 >> oversampling_setting);
+  if (variables.b7 < 0x80000000) {
+    variables.pressure = (variables.b7 * 2) / variables.b4;
+  } else {
+    variables.pressure = (variables.b7 / variables.b4) * 2;
+  }
+  variables.x1 = variables.pressure >> 8;
+  variables.x1 *= variables.x1;
+  variables.x1 = (variables.x1 * 3038) >> 16;
+  variables.x2 = (-7357 * variables.pressure) >> 16;
+  variables.pressure += (variables.x1 + variables.x2 + 3791) >> 4;
+
+  float pressure = static_cast<float>(variables.pressure);
+  pressure_results pressure_data = { .temperature =
+                                       temperature_data.temperature,
+                                     .pressure = pressure };
+  return pressure_data;
+}
+
+void bmp180::reset()
+{
+  hal::byte reset_setting = 0xB6;
+  std::array<hal::byte, 2> sensor_config_buffer = { soft_reset_register()[0],
+                                                    reset_setting };
```

Comment:
```suggestion
  hal::byte reset_setting = 0xB6;
  std::array<hal::byte, 2> sensor_config_buffer = { soft_reset_register()[0],
                                                    reset_setting };
```

Since the reset_setting value is actually a constant, consider having soft_reset_register() return back an array of 2 bytes with the reset setting in the right place. Then this call will just be a single `hal::write` with soft_reset_register in place of sensor_config_buffer.